cpp/src/gandiva/string_function_holder.cc

```diff
@@ -132,4 +133,33 @@ Status LikeHolder::Make(const std::string& sql_pattern, const std::string& escap
   return Status::OK();
 }
 
+Status ExtractHolder::Make(const FunctionNode& node,
+                           std::shared_ptr<ExtractHolder>* holder) {
+  ARROW_RETURN_IF(node.children().size() != 3,
+                  Status::Invalid("'extract' function requires three parameters"));
+
+  auto literal = dynamic_cast<LiteralNode*>(node.children().at(1).get());
+  ARROW_RETURN_IF(
+      literal == nullptr,
```

Comment:
Is possible to avoid code duplication doing:
ARROW_RETURN_IF( literal == nullptr || !IsArrowStringLiteral(literal->return_type()->id()), ...)

Reply:
The changes were applied